src/platform/plugins/shared/discover/public/ebt_manager/scoped_discover_ebt_manager.ts

```diff
@@ -150,6 +186,107 @@ export class ScopedDiscoverEBTManager {
     });
   }
 
+  private async trackFieldUsageInQueryEvent({
+    eventName,
+    fieldNames,
+    fieldsMetadata,
+  }: {
+    eventName: FieldUsageInQueryEventName;
+    fieldNames: string[];
+    fieldsMetadata: FieldsMetadataPublicStart | undefined;
+  }) {
+    if (!this.reportEvent) {
+      return;
+    }
+
+    const isOnlyFullTextSearch = fieldNames.length === 1 && fieldNames[0] === FREE_TEXT;
+
+    const eventData: FieldUsageInQueryEventData = {
+      [FIELD_USAGE_EVENT_NAME]: eventName,
+    };
+
+    // Handle full text search vs field-specific queries
+    if (isOnlyFullTextSearch) {
+      eventData[QUERY_FIELDS_USAGE_FIELD_NAMES] = [FREE_TEXT];
+    } else if (fieldsMetadata) {
+      // Process actual field names
+      const fields = await this.getFieldsFromMetadata({
+        fieldsMetadata,
+        fieldNames,
+      });
+
+      // tracks ECS compliant fields with a field name and non-ECS compliant fields with a "<non-ecs>" label
+      const categorizedFields = fieldNames.map((fieldName) =>
+        fields[fieldName]?.short ? fieldName : NON_ECS_FIELD
+      );
+
+      eventData[QUERY_FIELDS_USAGE_FIELD_NAMES] = [...new Set(categorizedFields)];
+    }
+
+    this.reportEvent(QUERY_FIELDS_USAGE_EVENT_TYPE, eventData);
+  }
+
+  public async trackSubmittingQuery({
+    query,
+    fieldsMetadata,
+  }: {
+    query: Query | AggregateQuery | undefined;
+    fieldsMetadata: FieldsMetadataPublicStart | undefined;
+  }) {
+    if (!query) {
+      return;
+    }
+
+    if (isOfAggregateQueryType(query)) {
+      // ES|QL query
+
+      if (query.esql.trim() === '') {
+        return;
+      }
+
+      const esqlColumns = getQueryColumnsFromESQLQuery(query.esql);
+      const embeddedQueries = getSearchQueries(query.esql); // KQL or Lucene embedded within ES|QL query
+
+      const embeddedQueryColumns = embeddedQueries
+        ? embeddedQueries
+            .map((embeddedQuery) => getKqlFieldNamesFromExpression(embeddedQuery))
+            .flat()
+        : [];
+
+      const fieldNames = [...esqlColumns, ...embeddedQueryColumns];
+
+      if (fieldNames.length === 0) {
+        return;
+      }
+
+      await this.trackFieldUsageInQueryEvent({
+        eventName: FieldUsageInQueryEventName.esqlQuery,
+        fieldNames,
+        fieldsMetadata,
+      });
+    } else if (isOfQueryType(query)) {
+      // KQL or Lucene query
+
+      if (typeof query.query !== 'string' || query.query.trim() === '') {
+        return;
+      }
+
+      const extractedFieldNames = getKqlFieldNamesFromExpression(query.query);
```

Comment:
One thing I noticed while testing is that when a wildcard is used against a field name in a KQL query (e.g. `byt* > 500`), we fail to track the resolved fields since this util just returns `byt*`. I'm ok treating this as a known limitation if it's not feasible, but it's something KQL already has to handle when generating the ES query. I'd suggest first checking with @lukasolson to see if it's easy to do, and if it's not then we can skip it for now.

Reply:
Thanks both! I'll consider it as a known limitation for this issue then. One followup question: in this case do we want to filter out any fields containing * or we want to track it as it is now (eg. "byt*" as a fieldName)?

Reply:
Ah that's true, those will result as non-ecs! Thx